lib/runner.js

```diff
@@ -429,7 +429,7 @@ Runner.prototype.runTest = function (fn) {
   if (!test) {
     return;
   }
-  if (this.forbidOnly && this.hasOnly) {
+  if (this.forbidOnly && hasOnly(this.parents().reverse()[0])) {
```

Comment:
...Just realized this *still* isn't right, because it won't work for tests outside of any `describe`. Remind me to add a test for that, but first, gonna fix this and merge...